interface/src/app/analyze/analysis/sampleBin.js

```diff
@@ -161,17 +161,25 @@ MathFuncts, errGen) {
       );
       // (2a) next, we build a new array (`retval`) comprised of
       //      `signatureObject`s by walking through the `firstSampleSignatures`
-      //       and constructing a signatureObject` for each. [outer .map()]
-      var retval = firstSampleSignatures.map(function(val) {
+      //      and constructing a `signatureObject` for each. [outer .map()]
+      var retval = firstSampleSignatures.map(function(val, index) {
         var signatureObject = {
           'id': val.node,
           'activity': this.heatmapData.samples.map(
             // (2b) the array of activity for each signature is built by
-            //      plucking the activity `.value` for each sample within this
-            //      signature from the `activityCache` [inner .map()]
+            //      plucking the activity `.value` for each sample within the
+            //      `index`th signature from the `activityCache` [inner .map()]
             function(sampleId) {
-              // FIXME: counting on array order to match signature order here
-              return this.activityCache.get(sampleId)[val.node - 1].value;
+              var cachedActivity = this.activityCache.get(sampleId);
+              if (cachedActivity[index].node !== val.node) {
+                // ensure we're pulling out the right node (aka signature)
+                $log.error(
+                  'getSignatureObjects: signature IDs do not match. First ' +
+                  ' sample = ', val, ', but sample ' + sampleId + ' =',
+                  cachedActivity[index]
```

Comment:
I did that on purpose. It's true that we're just checking the IDs to be sure they match, but if they *don't* match then I think I'll want to see the extra information to help diagnose what is going wrong. I expect this to be a rare error condition, so if it does arise I want it to be easy to track down the source of trouble.